sys/fmt/fmt.c

```diff
@@ -183,12 +183,12 @@ size_t fmt_u16_dec(char *out, uint16_t val)
 
 size_t fmt_s32_dec(char *out, int32_t val)
 {
-    int negative = (val < 0);
+    size_t negative = (val < 0 ? 1 : 0);
```

Comment:
and why ternary here - to be more explicit? If so I'd rather write it as `(val < 0) ? 1 :0;